### ISSUE TITLE
User can add a ticket directly to cart from the All Products block

### ISSUE THREAD

**Issue by ralucaStan** [woocommerce/woocommerce-gutenberg-products-block#5895]
## Describe the bug

In the All Product blocks ( and other blocks using the product grid), when clicking on the `Ticket details` button of a ticket type product,  the product is added directly to the cart.

## To reproduce

1. Make sure you have [woocommerce-box-office](https://github.com/woocommerce/woocommerce-box-office) extension installed
2. Create a ticket product
3. Add the All products block to a page
4. Go to the page with the All products block
5. Click on the Ticket details button of the Ticket product in the products grid
6. Notice that the product is added to the cart
7. Go to the Cart block/ shortcode and notice that the ticket quantity can be edited.

## Expected behavior

The button Ticket details should sent the user the the product's page, without adding the product to the cart 
The Ticket should no be editable in the Cart block.

## Screenshots

The Ticket product from the All products block.

<img width="192" alt="Screenshot 2022-02-18 at 17 33 16" src="https://user-images.githubusercontent.com/1628454/154723941-e6fbe562-9898-4c0a-a94d-416b77b52eb8.png">
For example, in the image below, the 1st ticket comes from the All Products block, while the second one, with the quantity of 2, was added from the product page. 
<img width="882" alt="Screenshot 2022-02-15 at 19 07 41" src="https://user-images.githubusercontent.com/1628454/154724106-93b1cf52-49c0-4227-8303-fe3b965ccc05.png">

**Comment by opr** [comment]
Thanks for reporting this Raluca!

It looks like we decide the action when someone presses the add to cart button based on these criteria:

https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/44b633957a5356414b93903b568aca81d83e3d6b/assets/js/atomic/blocks/product-elements/button/block.js#L69

We could possibly add an extensibility point here to control this, or we could look at the URL for the product, see if it's the same as the current URL, and decide then whether to do AJAX add to cart or redirect. It looks like that's how it works in the "Core shop page".

If you go to the box office plugin and change this:

https://github.com/woocommerce/woocommerce-box-office/blob/a930d8e311c0574ea8019fcd1c4b829fc4e9f7df/includes/class-wc-box-office-cart.php#L198-L203

to simply return `$cart` then pressing the button on the core shop page it will add it to the cart when pressing the button, rather than redirecting to the shop page.

To summarise we can either:

- Add extensibility point to allow extensions to change the add to cart functionality
- Check if the URL provided for the add to cart button matches the current page, and if not, redirect to the URL provided, if so, add via ajax.

**Comment by ralucaStan** [comment]
I wasn't able to replicate this anymore
